src/replay/edge_cases.zig

```diff
@@ -51,6 +51,11 @@ pub fn processEdgeCases(
     // have the wrong version. Our version was dead or pruned.
     // Signalled by ancestor_hashes_service.
     timer.reset();
+    // TODO: should we be holding the lock here? or pass the RwMux to the individual functions
+    // and have the locks be aquired at the site where they are accessed?
```

Comment:
Personally I would have just acquired the locks in the replay service's main orchestration function, and passed down const pointers to execution, consensus, and edge cases code. That keeps things less encumbered by synchronization code. You don't need to be judicious with read locks at all when you're in the same thread that is also the only thread that acquires write locks.

Reply:
[8c66e61](https://github.com/Syndica/sig/pull/875/commits/8c66e61293c912f1cd449b28a8c2b523173f5a19)